mitiq/cdr/cdr_tests.py

```diff
@@ -1,121 +1,134 @@
         gate = circ.data[i][0]
         name = gate.name
         if name == 'cx':
             qubit = [circ.data[i][1][0].index, circ.data[i][1][1].index]
-            parameters = []
             circ_new.cx(qubit[0], qubit[1])
-        if name =='rz':
-            parameters = (float(gate.params[0])) % (2 * np.pi)
-            # leave out empty Rz gates:
-            if parameters!=0:
-                qubit = circ.data[i][1][0].index
-                circ_new.rz(parameters, qubit)
-        if name =='sx':
-            parameters = np.pi/2
-            qubit = circ.data[i][1][0].index
-            circ_new.rx(parameters, qubit)
-        if name =='x':
+        elif name == 'u3':
             qubit = circ.data[i][1][0].index
-            circ_new.x(qubit)
-        elif name =='measure':
+            parameters = gate.params
+            # doing the decompostion of u3 gate here:
+            theta = parameters[0]
+            phi = parameters[1]
+            lamda = parameters[2]
+            circ_new.rz(phi+np.pi, qubit)
+            circ_new.rx(np.pi/2, qubit)
+            circ_new.rz(theta+np.pi, qubit)
+            circ_new.rx(np.pi/2, qubit)
+            circ_new.rz(lamda, qubit)
+        elif name == 'measure':
             qubit = circ.data[i][1][0].index
             cbit = circ.data[i][2][0].index
-            circ_new.measure(qubit, cbit) 
+            circ_new.measure(qubit, cbit)
     return(circ_new)
 
-def count_non_cliffords(
-    circuit: QPROGRAM,
-) -> float:
-    '''Function to check how many non-Clifford gates are in a give circuit.
-    
-    Args: 
-        circuit: some quantum circuit.
-        
-    Returns: 
-        number of non-Clifford gates in the given circuit.
-    '''
-    data, _ = circuit_to_array(circuit)
-    mask_rz = data[1, :] == 'rz'
-    rz_circ_data = data[:, mask_rz]
-    mask_not_rz = data[1, :] != 'rz'
-    not_rz_circ_data = data[:, mask_not_rz]
-    mask_non_cliff = is_clifford_angle(rz_circ_data[2, :]) == False
-    rz_non_cliff = rz_circ_data[:,mask_non_cliff]
-    return(len(rz_non_cliff[0]))
-
-num_qubits =4
-layers = 1
-measure = True
-num_training_circuits = 2
-fraction_non_clifford = 0.5
-circuit = cirq.circuits.Circuit(random_circuit(num_qubits,layers, measure))
-circuit = from_qiskit(qiskit_circuit_transplation(to_qiskit(circuit)))
+
+def uniform_circuit(
+) -> QPROGRAM:
+    """Returns a single qubit circuit with a uniform distribution of theta
+    rotations about the z axis.
+    Returns:
+        circ: qiskit quantum circuit object.
+    """
+    circ = QuantumCircuit(1, 1)
+    for i in range(2001):
+        angle = (i*np.pi*4)/(2000)
+        circ.rz(angle, 0)
+    circ.measure(0, 0)
+    return(circ)
+
+
+num_training_circuits = 10
+fraction_non_clifford = 0.90
+# num_qubits = 8
+# layers = 8
+# circuit = cirq.circuits.Circuit(random_circuit(num_qubits, layers))
+# circuit = from_qiskit(qiskit_circuit_transplation(to_qiskit(circuit)))
+circuit = from_qiskit(uniform_circuit())
```

Comment:
There shouldn't be any code outside unit tests. See https://github.com/unitaryfund/mitiq/blob/master/mitiq/tests/test_collector.py for an example. Basically, you want to define a series of tests `def test_something` which `assert`s basic facts that should be true about code in `clifford_training.data.py`.

Reply:
The code here looks like more of an example which could be included in docs or somewhere similar. Generally you don't want to be plotting things in unit tests.